app/src/main/java/db/TimelogEntryModel.java

```diff
@@ -0,0 +1,138 @@
+package db;
+
+import android.content.ContentResolver;
+import android.content.ContentUris;
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.BaseColumns;
+import android.util.Log;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.util.Date;
+
+public class TimelogEntryModel {
+  private static ContentResolver contentResolver;
+  private Uri uri;
+  private long id;
+  private Date startDate;
+  private Date endDate;
+  private String comment;
+
+  private DateFormat _UI_DATE_FORMATTER = DateFormat.getDateTimeInstance(
+      DateFormat.SHORT,
+      DateFormat.SHORT
+  );
+
+  public static void setContentResolver( ContentResolver resolver ) {
+    TimelogEntryModel.contentResolver = resolver;
+  }
+
+  public TimelogEntryModel( Date startDate, Date endDate, String comment ) {
+    Log.d( "Model", "Value constructor called");
+    this.id = -1;
+    this.uri = null;
+    this.startDate = startDate;
+    this.endDate = endDate;
+    this.comment = comment;
+  }
+
+  private TimelogEntryModel() {}
+
+  // Assumes cursor is not null and points to the correct provider record
+  public TimelogEntryModel( final Cursor cursor ) throws ParseException {
+    Log.d( "Model", "Cursor constructor called");
+    final int idColumnIndex = cursor.getColumnIndex( BaseColumns._ID );
```

Comment:
Ja. Wenn man explizit bei `projection` die Spalten angibt, kommen diese in derselben Reihenfolge im Ergebnis vor.